platforms/cpuinfo.go

```diff
@@ -98,7 +101,11 @@ func getCPUVariant() string {
 	variant, err := getCPUInfo("Cpu architecture")
```

Comment:
Possibly for a follow-up, but I'd be interested to see a benchmark comparison between `getCPUInfo()` and the new `getCPUArchitecture()`

1. do we know if `getCPUArchitecture()` can provide all the information that the existing `getCPUInfo()` can provide? Or are there (corner-case) situations where `getCPUInfo()` can still provide details that `getCPUArchitecture()` cannot provide?
2. perhaps it can provide more (as it doesn't depend on `/proc/cpuinfo`, so potentially can be used on non-Linux platforms https://github.com/containerd/containerd/pull/7295/files#r947149903
3. `getCPUInfo()` has to parse `/proc/cpuinfo`, which may be more resource-intensive as the new function (however, `getCPUVariant()` is already called with a `sync.Once`, so probably not a big deal)


Depending on the above, we could consider;

- swapping the order of `getCPUArchitecture()` and `getCPUInfo()`
- removing `getCPUInfo()` altogether (if `getCPUArchitecture()` can provide the same information

Reply:
No, `getCPUArchitecture` cannot provide all the information of `getCPUInfo`, `getCPUArchitecture` only gets `Cpu Architecture` information, but `getCPUInfo` can get a lot of information about CPU, such as `model name`, `cpu MHz`, `cache size`, etc.


It is not recommended to adjust the positions of `getCPUInfo` and `getCPUArchitecture`, because `getCPUInfo` seems to work well in most cases. At present, most of the problems are because `qemu` is used on mac to simulate `linux/arm64` platform, resulting in `/proc/cpuinfo` file does not exist `Cpu architecture` information is wrong.

such as: 
```
$ docker run -ti --rm --platform linux/arm64 alpine:3.16.1
/ # cat /proc/cpuinfo  | grep architecture
/# no result
```

So `getCPUArchitecture` is only used when the `getCPUInfo` method gets an error.

Reply:
> At present, most of the problems are because qemu is used on mac to simulate linux/arm64 platform, resulting in /proc/cpuinfo file does not exist Cpu architecture information is wrong.

Some initial thoughts on that; first of all, the userland emulation with QEMU is a "best effort" attempt on doing emulation; running containerd with emulation ("docker-in-docker") may still reveal corner case situations (even more taking into account that containers started with it may now also be running with QEMU emulation).

- Are you seeing different output of `/proc/cpuinfo` with and without QEMU emulation? i.e., does the output differ with running the same command, but with the actual host's architecture? (`docker run --rm --platform linux/amd64 alpine cat /proc/cpuinfo`)
- For the userland QEMU _emulation_ (non-matching architecture), the containerd binary should have `GOARCH` matchin the emulated platform.
- For the ARM case where `getCPUInfo()` fails, I wonder if we should apply a similar switch as currently is in place at the top (macOS, Windows), and use `v8` as default for `GOARCH==aarch64` instead of returning early; https://github.com/containerd/containerd/blob/77d53d2d230c3bcd3f02e6f493019a72905c875b/platforms/cpuinfo.go#L85-L93

Reply:
- "Are you seeing different output of /proc/cpuinfo with and without QEMU emulation?" , yes；eg:
   1. In the mac `docker run --rm --platform linux/amd64 alpine cat /proc/cpuinfo | grep "arch"` is not any print.
   2. In the Android run `cat /proc/cpuinfo | grep "arch"`  is print result.(use `Termux` tools)
- current `runtime.GOARCH`  in the goland only having `arm,arm64,arm64be,armbe` this value, not arm variant info.
- `runtime.GOARCH` not `aarch64` value.